### ISSUE TITLE
Descirbe images with Qwen2.5-VL-3B-Instruct-MNN

### ISSUE THREAD

**Issue by KevinZhang2023** [alibaba/MNN#3326]
Thank you for your attention.
With the following script, I am trying to describe an image with MNN/Qwen2.5-VL-3B-Instruct-MNN. But I get the following error.

gdb ./llm_cpp_demo

(gdb) run
Starting program: /root/qwen/mnn/llm_cpp_demo 
warning: Error disabling address space randomization: Operation not permitted
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
The device supports: i8sdot:0, fp16:0, i8mm: 0, sve2: 0, sme2: 0
[New Thread 0x7f3503f62000 (LWP 8270)]
[New Thread 0x7f3503761000 (LWP 8271)]
[New Thread 0x7f3502f60000 (LWP 8272)]
=== Model Input Info ===
Input Name: attention_mask
  Shape: 1 1 -1 -1 
  Data Type: 2
Input Name: input_ids
  Shape: 1 1 2048 
  Data Type: 2
Input Name: past_key_values
  Shape: 36 2 1 -1 2 128 
  Data Type: 2
Input Name: position_ids
  Shape: 1 -1 
  Data Type: 0

[New Thread 0x7f34f8bf2000 (LWP 8291)]
[New Thread 0x7f34f87f1000 (LWP 8292)]
Compute Shape Error for /rotary/Cast_output_0
Invalid Tensor, the session may not be ready
Reshape error: 4 -> 0
Compute Shape Error for /Reshape_output_0

Thread 1 "llm_cpp_demo" received signal SIGSEGV, Segmentation fault.
__memmove_evex_unaligned_erms () at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:396
396     ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S: No such file or directory.
(gdb) bt
#0  __memmove_evex_unaligned_erms () at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:396
#1  0x00005556bf8d4efe in run_inference (config_path="/root/qwen/mnn/config.json", image_path="/root/qwen/thermometer.png", prompt="Describe the image") at main.cpp:178
#2  0x00005556bf8d5727 in main () at main.cpp:215

What is the problem? Thank you for your time.
---------------------------------------------------------
#include <iostream>
#include <string>
#include <vector>
#include <map>
#include <fstream>
#include <opencv2/opencv.hpp>
#include <MNN/Interpreter.hpp>
#include <MNN/ImageProcess.hpp>
#include <MNN/Tensor.hpp>
#include <jsoncpp/json/json.h>
#include <cstdint> // Include for int32_t

using namespace MNN;
using namespace std;

cv::Mat preprocess_image(const string& path, int target_size = 448) {
    cv::Mat img = cv::imread(path);
    if (img.empty()) {
        cerr << "Error loading image: " << path << endl;
        return {};
    }

    // Convert BGR to RGB
    cv::cvtColor(img, img, cv::COLOR_BGR2RGB);

    // Resize with aspect ratio preservation
    cv::Mat resized_img;
    int orig_w = img.cols, orig_h = img.rows;
    float ratio = min(static_cast<float>(target_size)/orig_w,
                     static_cast<float>(target_size)/orig_h);
    cv::resize(img, resized_img, cv::Size(round(orig_w*ratio), round(orig_h*ratio)));

    // Pad to square
    int pad_w = target_size - resized_img.cols;
    int pad_h = target_size - resized_img.rows;
    cv::copyMakeBorder(resized_img, resized_img,
                      0, pad_h, 0, pad_w,
                      cv::BORDER_CONSTANT, cv::Scalar(0,0,0));

    // Convert to float32 and normalize
    resized_img.convertTo(resized_img, CV_32FC3, 1.0/255.0);

    return resized_img;
}

vector<int> tokenize_text(const string& text, int max_length = 512) {

    return {101, 234, 345, 567};
}

string run_inference(const string& config_path,
                    const string& image_path,
                    const string& prompt) {

    ifstream config_file(config_path);
    if (!config_file.is_open()) {
        cerr << "Error opening config file: " << config_path << endl;
        return "";
    }

    Json::Value config;
    if (!Json::Reader().parse(config_file, config)) {
        cerr << "Error parsing config file" << endl;
        return "";
    }

    auto interpreter = shared_ptr<Interpreter>(Interpreter::createFromFile(config["llm_model"].asCString()));
    if (!interpreter) {
        cerr << "Error creating interpreter" << endl;
        return "";
    }

    ScheduleConfig mnn_config;
    mnn_config.type = MNN_FORWARD_CPU;
    if (config.isMember("thread_num")) {
        mnn_config.numThread = config["thread_num"].asInt();
    }
    BackendConfig backend_config;
    backend_config.precision = BackendConfig::Precision_Low;
    backend_config.memory = BackendConfig::Memory_Low;
    mnn_config.backendConfig = &backend_config;

    Session* session = interpreter->createSession(mnn_config);
    if (!session) {
        cerr << "Error creating session" << endl;
        return "";
    }

    auto input_info = interpreter->getSessionInputAll(session);
    if (input_info.empty()) {
        cerr << "Error: No input tensors found" << endl;
        return "";
    }

    cout << "=== Model Input Info ===" << endl;
    vector<string> input_names;
    for (auto& pair : input_info) {
        input_names.push_back(pair.first);
        cout << "Input Name: " << pair.first << endl;
        cout << "  Shape: ";
        for (auto d : pair.second->shape()) cout << d << " ";
        cout << "\n  Data Type: " << pair.second->getType().code << endl;
    }

    auto output_info = interpreter->getSessionOutputAll(session);
    if (output_info.empty()) {
        cerr << "Error: No output tensors found" << endl;
        return "";
    }

    if (input_names.size() < 2) {
        cerr << "Error: Model requires at least 2 inputs" << endl;
        return "";
    }

    const string& image_input_name = input_names[0];
    const string& text_input_name = input_names[1];

    Tensor* image_input = interpreter->getSessionInput(session, image_input_name.c_str());
    Tensor* text_input = interpreter->getSessionInput(session, text_input_name.c_str());
    if (!image_input || !text_input) {
        cerr << "Error: Failed to get input tensors" << endl;
        return "";
    }

    cv::Mat image = preprocess_image(image_path);
    if (image.empty()) return "";

    MNN::CV::ImageProcess::Config img_config;
    img_config.filterType = MNN::CV::BILINEAR;
    img_config.sourceFormat = MNN::CV::RGB;
    img_config.destFormat = MNN::CV::RGB;

    float mean_values[4] = {0.485f, 0.456f, 0.406f, 0.0f};
    float norm_values[4] = {0.229f, 0.224f, 0.225f, 1.0f};
    for(int i=0; i<4; ++i) {
        img_config.mean[i] = mean_values[i];
        img_config.normal[i] = norm_values[i];
    }

    std::shared_ptr<MNN::CV::ImageProcess> processor(MNN::CV::ImageProcess::create(img_config));

    interpreter->resizeTensor(image_input, {1, 3, 448, 448});
    interpreter->resizeSession(session);

    processor->convert(image.data, 448, 448, 0, image_input);

    vector<int> tokens = tokenize_text(prompt);
    if (tokens.empty()) {
        cerr << "Error: Tokenization failed" << endl;
        return "";
    }

    interpreter->resizeTensor(text_input, {1, static_cast<int>(tokens.size())});
    interpreter->resizeSession(session);

    // **Explicitly use int32_t for memcpy:**
    memcpy(text_input->host<int32_t>(), tokens.data(), tokens.size() * sizeof(int32_t)); // Changed to int32_t

    auto status = interpreter->runSession(session);
    if (status != NO_ERROR) {
        cerr << "Inference failed with code: " << status << endl;
        return "";
    }

    Tensor* output_tensor = interpreter->getSessionOutput(session, output_info.begin()->first.c_str());
    if (!output_tensor) {
        cerr << "Error: Failed to get output tensor" << endl;
        return "";
    }

    const uint8_t* output_data = output_tensor->host<uint8_t>();
    size_t output_size = output_tensor->elementSize();

    string result;
    try {
        result = string(reinterpret_cast<const char*>(output_data), output_size);
    } catch (const std::exception& e) {
        cerr << "Error converting output: " << e.what() << endl;
        return "";
    }

    interpreter->releaseSession(session);
    return result;
}

int main() {
    string config = "/root/qwen/mnn/config.json";
    string image = "/root/qwen/thermometer.png";
    string prompt = "Describe the image.";

    string response = run_inference(config, image, prompt);
    if (!response.empty()) {
        cout << "=== Model Response ===" << endl;
        cout << response << endl;
    } else {
        cerr << "Inference returned empty result" << endl;
    }

    return 0;
}

**Comment by jxt1234** [comment]
Qwen2.5-VL-3B-Instruct-MNN must use mnn-llm to load and execute.
See https://mnn-docs.readthedocs.io/en/latest/transformers/llm.html for detail.

**Comment by KevinZhang2023** [comment]
Solved. Thanks.
